controller_manager:
  ros__parameters:

    update_rate: 25

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    manipulator_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    forward_position_controller:
      type: position_controllers/JointGroupPositionController


manipulator_controller:
  ros__parameters:
    joints:
      - J1
      - J2
      - J3
      - J4
      - J5
      - J6
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    open_loop_control: true
    # state_publish_rate: 100.0
    # action_monitor_rate: 20.0
    # allow_partial_joints_goal: false
    constraints:
    #   stopped_velocity_tolerance: 0.2
      goal_time: 0.0
      # J1: { trajectory: 0.02, goal: 0.01 }
      # J2: { trajectory: 0.02, goal: 0.01 }
      # J3: { trajectory: 0.02, goal: 0.01 }
      # J4: { trajectory: 0.02, goal: 0.01 }
      # J5: { trajectory: 0.02, goal: 0.01 }
      # J6: { trajectory: 0.02, goal: 0.01 }


forward_position_controller:
  ros__parameters:
    joints:
      - J1
      - J2
      - J3
      - J4
      - J5
      - J6